Onboarding/Tests/Specs/OnboardingContainerViewModelTests.swift

```diff
@@ -0,0 +1,273 @@
+import XCTest
+import Combine
+
+@testable import Onboarding
+
+final class OnboardingContainerViewModelTests: XCTestCase {
+
+    private var sut: OnboardingContainerViewModel?
+    private var cancellables = Set<AnyCancellable>()
+
+    func test_init_hasCorrectInitialState() throws {
+        let sut = OnboardingContainerViewModel(
+            onboardingService: MockOnboardingService(),
+            onboardingType: .json("test"),
+            dismissAction: {}
+        )
+
+        XCTAssertEqual(sut.state, .loading)
+        XCTAssertEqual(sut.tabIndex, 0)
+        XCTAssertEqual(sut.skipButtonTitle, "Skip")
+        XCTAssertEqual(sut.skipButtonAcessibilityHint, "Skip onboarding")
+    }
+
+    func test_init_fetchedSlides_changesState() throws {
+        let mockOnboardingService = MockOnboardingService()
+
+        let expectedResource = "MockOnboardingResponse"
+        let expectedSlides = [
+            OnboardingSlide(image: "test", title: "test_title", body: "test_body"),
+            OnboardingSlide(image: "test1", title: "test_title", body: "test_body")
+        ]
+
+        let sut = OnboardingContainerViewModel(
+            onboardingService: mockOnboardingService,
+            onboardingType: .json(expectedResource),
+            dismissAction: {}
+        )
+        let expectation = XCTestExpectation(description: "Slide return")
+        sut.$state
+            .dropFirst()
+            .sink(receiveValue: {
+                XCTAssertEqual($0, .loaded(expectedSlides))
+                expectation.fulfill()
+            })
+            .store(in: &cancellables)
+
+        mockOnboardingService._receivedFetchSlidesCompletionHander?(.success(expectedSlides))
+
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func test_init_fetchedSlides_emptySlides_completesFlow() throws {
+        let mockOnboardingService = MockOnboardingService()
+
+        let expectedResource = "MockOnboardingResponse"
+
+        let expectation = XCTestExpectation(description: "Empty slides")
+        sut = OnboardingContainerViewModel(
+            onboardingService: mockOnboardingService,
+            onboardingType: .json(expectedResource),
+            dismissAction: {
+                expectation.fulfill()
+            }
+        )
+
+        mockOnboardingService._receivedFetchSlidesCompletionHander?(.success([]))
+
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func test_init_jsonType_fetchesSlides() throws {
```

Comment:
jsonSource or just json if we dont want to worry about renaming if we change the OnboardingSource class name

Reply:
changed to json